stdlib/SparseArrays/src/linalg.jl

```diff
@@ -327,11 +327,126 @@ function bwdTriSolve!(A::SparseMatrixCSCUnion, B::AbstractVecOrMat)
     B
 end
 
+fwdTriSolve!(aA::Adjoint{<:Any,<:SparseMatrixCSCUnion}, B::AbstractVecOrMat) =
+    _fwdTriSolve!(aA.parent, B, true)
+
+fwdTriSolve!(aA::Transpose{<:Any,<:SparseMatrixCSCUnion}, B::AbstractVecOrMat) =
+    _fwdTriSolve!(aA.parent, B, false)
+
+function _fwdTriSolve!(A::SparseMatrixCSCUnion, B::AbstractVecOrMat, adj::Bool)
+# forward substitution for adjoints of CSC matrices
+    @assert !has_offset_axes(A, B)
+    nrowB, ncolB  = size(B, 1), size(B, 2)
+    ncol = LinearAlgebra.checksquare(A)
+    if nrowB != ncol
+        throw(DimensionMismatch("A is $(ncol) columns and B has $(nrowB) rows"))
+    end
+
+    aa = getnzval(A)
+    ja = getrowval(A)
+    ia = getcolptr(A)
+
+    joff = 0
+    for k = 1:ncolB
+        for j = 1:nrowB
+            i1 = ia[j]
+            i2 = ia[j + 1] - 1
+            akku = B[joff + j]
+            done = false
+
+            # loop through column j of A - only structural non-zeros
+            for ip = i1:i2
+                i = ja[ip]
+                aai = adj ? aa[ip]' : aa[ip]
+                if i < j
+                    akku -= B[joff + i] * aai
+                elseif i == j
+                    B[joff + j] = akku / aai
+                    done = true
+                    break
+                end
+            end
+            if !done
+                throw(LinearAlgebra.SingularException(j))
+            end
+        end
+        joff += nrowB
+    end
+    B
+end
+
+bwdTriSolve!(aA::Adjoint{<:Any,<:SparseMatrixCSCUnion}, B::AbstractVecOrMat) =
+    _bwdTriSolve!(aA.parent, B, true)
+
+bwdTriSolve!(aA::Transpose{<:Any,<:SparseMatrixCSCUnion}, B::AbstractVecOrMat) =
+    _bwdTriSolve!(aA.parent, B, false)
+
+function _bwdTriSolve!(A::SparseMatrixCSCUnion, B::AbstractVecOrMat, adj::Bool)
+# forward substitution for adjoints of CSC matrices
+    @assert !has_offset_axes(A, B)
+    nrowB, ncolB  = size(B, 1), size(B, 2)
+    ncol = LinearAlgebra.checksquare(A)
+    if nrowB != ncol
+        throw(DimensionMismatch("A is $(ncol) columns and B has $(nrowB) rows"))
+    end
+
+    aa = getnzval(A)
+    ja = getrowval(A)
+    ia = getcolptr(A)
+
+    joff = 0
+    for k = 1:ncolB
+        for j = nrowB:-1:1
+            i1 = ia[j]
+            i2 = ia[j + 1] - 1
+            akku = B[joff + j]
+            done = false
+
+            # loop through column j of A - only structural non-zeros
+            for ip = i2:-1:i1
+                i = ja[ip]
+                aai = adj ? aa[ip]' : aa[ip]
+                if i > j
+                    akku -= B[joff + i] * aai
+                elseif i == j
+                    B[joff + j] = akku / aai
+                    done = true
+                    break
+                end
+            end
+            if !done
+                throw(LinearAlgebra.SingularException(j))
+            end
+        end
+        joff += nrowB
+    end
+    B
+end
+
+const LowerTriangular2{T} = Union{
```

Comment:
Probably not worth it if the union here is only used once

Reply:
Improved readability?

Reply:
I think it is easier just to read the union in the method definition since it is not clear what `LowerTriangular2` is.